### PR TITLE
Script to email users about price drops.

### LINKED ISSUE #20: Create an email.py script that notifies customers of discounts
# Description
Develop a Python script that notifies subscribed customers via email if a product's price drops below their specified discount threshold. Customers can set this threshold when subscribing to a product through the Streamlit dashboard. The script should check the current product price every 3 minutes and, if the price change surpasses the threshold, send an email to the customer's registered email address. Each customer should only be notified once per price drop.

# Outcome
- `email.py`: script that uses `boto3` to monitor product prices and send email alerts to subscribed users when a price surpasses their discount threshold. The script ensures:
  - Price checks occur every 3 minutes (part of the ETL pipeline).
  - Alert emails are sent only once per qualifying price change. 
- **boto3 Integration:** Use AWS's boto3 library to send emails via Amazon SES (Simple Email Service). 
- `test_email.py`: A test suite (if feasible) to validate the email functionality and ensure correctness, adhering to TDD practices.

# User Story
As a customer, I want to be able to receive email alerts about a product I have subscribed to if the price has been deducted and surpassed the discount threshold. I want to be alerted within 3 minutes of the change in price.

### PR DESCRIPTION
**Pull request by indiah444**
- Added necessary libraries (`boto3`) to `requirements.txt`
- Created `email.py` to send emails to users when the product they've subscribed to has dropped below their chosen threshold.

This (partly) closes #20, I will write some tests to follow this up shortly.

### CODE REVIEW

File: pipeline/email.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+from connect_to_database import configure_logging, get_connection, get_cursor
+
+
+def get_ses_client() -> boto3.client:
+    """Returns the boto3 SES client to send emails with."""
+
+    return boto3.client("ses", region_name="eu-west-2",
+                        aws_access_key_id=ENV["AWS_ACCESS_KEY_ID"],
+                        aws_secret_access_key=ENV["AWS_ACCESS_SECRET_KEY"])
```
Comment (gem09lo) [inline pipeline/email.py:18]: Optional! 
```suggestion
aws_access_key = ENV.get("AWS_ACCESS_KEY_ID")
aws_secret_key = ENV.get("AWS_ACCESS_SECRET_KEY")
from_email = ENV.get("FROM_EMAIL")

if not aws_access_key or not aws_secret_key or not from_email:
    raise RuntimeError("Required environment variables for AWS or email are missing")
```

**Review by gem09lo — COMMENTED** [review]
Overall its clear and easily understandable! Only had 2 comments!

File: pipeline/email.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+
+def get_subscriptions_and_products(conn: connection):
```
[the remark this answers is not in the document]
Comment (indiah444) [reply pipeline/email.py:21]: Yes will do, good spot!

File: pipeline/email.py (hunk 1 above)
Comment (indiah444) [reply pipeline/email.py:18]: Ooh, great idea I like that! I'll implement this now :)

**Pull request merged**